### ISSUE TITLE
Nullability of `TupleN` components

### ISSUE THREAD

**Issue by dynaxis** [reactor/reactor-kotlin#5]
According to the documentation on `TupleN.get(int)`, `TupleN` components may be `null`. So I put `T?` as the return type of `componentN()` operators on `Tuple` classes. But using them, in my project, I've found that having to check if they are `null` while knowing they are non-null is generally annoying in Kotlin.
So I suggest `TupleN.componentN()`s be all made to return a non-null value. And if a component may be `null`, then users are expected to use `TupleN.get(int)` instead where it is clearly marked as `@Nullable`.
What do you think, @sdeleuze?

**Comment by dynaxis** [comment]
As another option, we may omit the return type of the `componentN()` operators. In that case, the return type will be something like `T!` rather than `T` or `T?`, which means the return type is from Java and may be nullable. But Kotlin doesn't enforce null checks.
FYI, required changes are really trivial, but I plan to issue a PR.

**Comment by dynaxis** [comment]
Self resolving this issue. I got interactions between Kotlin's generics and nullability wrong. So my pull request for `reactor.util.funtion.TupleN`s should have set the return type of `componentN()`s to just `T` not `T?`. In that way, `TupleN` with nullable components work equally well.
And the above comment was wrong. Actually setting the return types to `T` doesn't actually differ from leaving it out to be inferred. But specifying `T` explicitly silences IDEs.
So I'll issue another PR. Sorry for the confusion. Closing this issue.
